tests/test_rds.py

```diff
@@ -923,6 +923,30 @@ def test_rds_db_option_groups(self):
         self.assertEqual(len(resources), 1)
         self.assertEqual(resources[0]["DBInstanceIdentifier"], "database-2")
 
+    def test_rds_pending_maintenance(self):
+        session_factory = self.replay_flight_data("test_rds_has_pending_maintenance")
+        p = self.load_policy(
+            {
+                "name": "rds-pending-maintenance",
+                "resource": "rds",
+                "filters": [
+                    {
+                        "type": "value",
+                        "key": "DBInstanceIdentifier",
+                        "value": "qbopp011"
```

Comment:
this is currently recording a lot of other db instances, which is causing a large +12k diff. there are typically four options here to clean up the recording. 

1. using a clean environment which only has the resource that's being tested
2. using a server side filter if the resource type supports one ('query' in a policy)
3. restructuring the test to just fetch one and pass that directly to the filter.
4. manually editing the recorded json.

Reply:
@sgungi-intuit the test file (tests/data/placebo/test_rds_pending_maintenance/rds.DescribePendingMaintenanceActions_1.json) still has too much unnecessary data. We should be able to clean it up even more. You should only need 2 items in there, one that matches the DB instance you're testing for and one that doesn't

Reply:
Hi @darrendao, the "describe_pending_maintenance_actions" call lists of all pending maintenances for all RDS instances, not sure if there is any query filter to limit to just one rds instance. 


        results = []
        pending_maintenance = set()
        paginator = client.get_paginator('describe_pending_maintenance_actions')
        for page in paginator.paginate():
            pending_maintenance.update(
                {action['ResourceIdentifier'] for action in page['PendingMaintenanceActions']}
            )

        for r in resources:
            if r['DBInstanceArn'] in pending_maintenance:
                results.append(r)

        return results

Reply:
@sgungi-intuit as Kapil has mentioned, you can use a clean environment or just edit the recorded json. I would just edit the recorded json. For example
```
{
    "status_code": 200,
    "data": {
      "PendingMaintenanceActions": [
        {
          "ResourceIdentifier": "arn:aws:rds:us-west-2:644160558196:db:qbopp010",
          "PendingMaintenanceActionDetails": [
            {
              "Action": "system-update",
              "Description": "New Operating System update is available"
            }
          ]
        },
        {
          "ResourceIdentifier": "arn:aws:rds:us-west-2:644160558196:db:qbopp011",
          "PendingMaintenanceActionDetails": [
            {
              "Action": "system-update",
              "Description": "New Operating System update is available"
            }
          ]
        }
      ],
      "ResponseMetadata": {}
    }
  }
```
I think that should work